lfs/gitscanner_catfilebatch.go

```diff
@@ -23,42 +23,41 @@ func runCatFileBatch(pointerCh chan *WrappedPointer, revs *StringChannelWrapper,
 		return err
 	}
 
-	go catFileBatchOutput(pointerCh, cmd, errCh)
-	go catFileBatchInput(cmd, revs, errCh)
-	return nil
-}
+	go func() {
+		scanner := &catFileBatchScanner{r: cmd.Stdout}
+		for r := range revs.Results {
+			cmd.Stdin.Write([]byte(r + "\n"))
+			hasNext := scanner.Scan()
```

Comment:
I'm a little confused by how this method works 😱 . My inclination would be that `Scan()` scans for a pointer, and returns whether or not a _subsequent_ scan will succeed.

However, from looking at the code below, it looks like it only returns whether the current scan was OK, and returns false even if we're at the end of the stream.

Should the scanner buffer two (or n) pointers instead of just one? If you end up implementing this, you'll have to be careful that `io.Reader`s can return EOF at the last read (n>0, io.EOF) or directly _after_ the last read (0, io.EOF).